### PR TITLE
Add MultiDeleteCache interface and default CacheProvider implementation

### PR DESCRIPTION
**Pull request by bburnichon**
I have a case where I need to invalidate several keys at once. This port the addition I made to provide it to all users of this library.

Also add memcached implementation.

### CODE REVIEW

File: lib/Doctrine/Common/Cache/MultiDeleteCache.php [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+ * @link   www.doctrine-project.org
+ * @since  1.7
+ * @author Benoit Burnichon <bburnichon@gmail.com>
+ */
+interface MultiDeleteCache
+{
+    /**
+     * Returns a boolean value indicating if the operation succeeded.
+     *
+     * @param array $keys Array of keys to delete from cache
```
Comment (Ocramius) [inline lib/Doctrine/Common/Cache/MultiDeleteCache.php]: `string[]`

File: lib/Doctrine/Common/Cache/MultiDeleteCache.php [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+ * Interface for cache drivers that allows to put many items at once.
+ *
+ * @link   www.doctrine-project.org
+ * @since  1.7
+ * @author Benoit Burnichon <bburnichon@gmail.com>
+ */
+interface MultiDeleteCache
+{
+    /**
+     * Returns a boolean value indicating if the operation succeeded.
```
Comment (Ocramius) [inline lib/Doctrine/Common/Cache/MultiDeleteCache.php]: Already documented in `@return`

File: lib/Doctrine/Common/Cache/CacheProvider.php [hunk 3]
```diff
@@ -135,6 +136,19 @@ public function save($id, $data, $lifeTime = 0)
     /**
      * {@inheritdoc}
      */
+    public function deleteMultiple(array $keys)
+    {
+        $namespacedKeys = array();
+        foreach ($keys as $key) {
+            $namespacedKeys[] = $this->getNamespacedId($key);
```
Comment (Ocramius) [inline lib/Doctrine/Common/Cache/CacheProvider.php]: `return $this->doDeleteMultiple(array_values(array_map([$this, 'getNamespacedId'], $keys)));`

Comment (Ocramius) [inline lib/Doctrine/Common/Cache/CacheProvider.php]: the `array_values` is not important if `doDeleteMultiple` doesn't care about the array keys.

**Comment by bburnichon** [comment]
ping @Ocramius @guilhermeblanco Is there anything wrong with this PR?

**Comment by Ocramius** [comment]
Simply didn't look at it yet, sorry!

**Comment by Ocramius** [comment]
@bburnichon aaand merged, thanks a lot!

**Pull request closed without merging**